### PR TITLE
[SPARK-32191][PYTHON][DOCS] Port migration guide for PySpark docs

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This proposes to port old PySpark migration guide to new PySpark docs.

### Why are the changes needed?

Better documentation.

### Does this PR introduce _any_ user-facing change?

No. Documentation only.

### How was this patch tested?

Generated document locally.

Screen Shot 2020-08-07 at 1 53 20 PM

### CODE REVIEW

File: python/docs/source/migration_guide/index.rst [hunk 1]
```diff
@@ -20,3 +20,14 @@
...
+Migration Guide: PySpark (Python on Spark)
+
+Note that this migration guide describes the items specific to PySpark.
+Many items of SQL migration can be applied when migrating PySpark to higher versions.
+Please refer `Migration Guide: SQL, Datasets and DataFrame <http://spark.apache.org/docs/latest/sql-migration-guide.html>`_.
+
+.. toctree::
+   :maxdepth: 2
+
+   pyspark_2.4_to_3.0
```
Comment (viirya) [inline python/docs/source/migration_guide/index.rst:32]: Question: do we want to move all old migration guide?

**Comment by viirya** [comment]
cc @HyukjinKwon

**Comment by HyukjinKwon** [comment]
Nice, thank you @viirya! I will take a look next Monday!

File: python/docs/source/migration_guide/index.rst (hunk 1 above)
Comment (HyukjinKwon) [reply python/docs/source/migration_guide/index.rst:32]: Oh yeah, I was thinking just completely move all, and just leave one link to redirect. I haven't checked how difficult to add a link back. If that's technically difficult, we can just guide users to read PySpark documentation without a link.

File: python/docs/source/migration_guide/pyspark_1.0_1.2_to_1.3.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+=====================================
+Upgrading from PySpark 1.0-1.2 to 1.3
+=====================================
+
+.. currentmodule:: pyspark_1.0_1.2_to_1.3
```
Comment (HyukjinKwon) [inline python/docs/source/migration_guide/pyspark_1.0_1.2_to_1.3.rst:23]: I believe these `currentmodule`s in this and other pages here can be removed.

File: python/docs/source/migration_guide/index.rst [hunk 3]
```diff
@@ -20,3 +20,18 @@
 Migration Guide
```
Comment (HyukjinKwon) [inline python/docs/source/migration_guide/index.rst:20]: I think now we can remove all the contents at https://github.com/apache/spark/blob/master/docs/pyspark-migration-guide.md (like we did at https://github.com/apache/spark/blob/master/docs/sql-migration-old.md in this commit https://github.com/apache/spark/commit/7d4eb38bbcc887fb61ba7344df3f77a046ad77f8)

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise. Might be good to update PR description before it gets merged.

File: python/docs/source/migration_guide/index.rst (hunk 3 above)
Comment (HyukjinKwon) [reply python/docs/source/migration_guide/index.rst:20]: The link of PySpark API documentation should be https://spark.apache.org/docs/latest/api/python/migration_guide/index.html

Comment (viirya) [reply python/docs/source/migration_guide/index.rst:20]: Do you mean the link below? The link is pointing readers to SQL migration, not PySpark API documentation.

Comment (HyukjinKwon) [reply python/docs/source/migration_guide/index.rst:20]: Ohhh no I mean, we can remove all contents in `pyspark-migration-guide.md` but keep one line such as:

```
The migration guide is now archived on [this page](https://spark.apache.org/docs/latest/api/python/migration_guide/index.html).
```

like [sql-migration-old.md](https://github.com/apache/spark/blob/master/docs/sql-migration-old.md).

Comment (viirya) [reply python/docs/source/migration_guide/index.rst:20]: Oh I see. Thanks.

**Comment by HyukjinKwon** [comment]
Merged to master. Thanks @viirya for working on this.

**Pull request closed without merging**